<launch>
    <!-- NOTE: Training images need to be in the VOC format. See readme for suggestions. -->
    <arg name="train_image_path" value="/home/diggerdata/datasets/coco/train2014/" />
    <arg name="train_annot_path" value="/home/diggerdata/datasets/coco/train2014ann/" />
   
    <!-- The following are optional. If not given will split the training set 0.8/0.2 - train/valid. -->
    <arg name="valid_image_path" value="/home/diggerdata/datasets/coco/val2014/" />
    <arg name="valid_annot_path" value="/home/diggerdata/datasets/coco/val2014ann/" />

    <node pkg="yolov2_ros" name="yolo_train" type="yolo_train.py" output="screen">
        <param name="n_gpu" type="int" value="2" />
        <param name="backend" type="str" value="full_yolo" />
        <param name="input_size" type="int" value="416" />
        <param name="max_number_detections" type="int" value="10" />
        <rosparam param="anchors">[0.57273, 0.677385, 1.87446, 2.06253, 3.33843, 5.47434, 7.88282, 3.52778, 9.77052, 9.16828]</rosparam>
        
        <param name="weights_path" type="str" value="$(find yolov2_ros)/weights" />
        <param name="weight_file" type="str" value="" />
        
        <param name="train_annot_folder" type="str" value="$(arg train_annot_path)" />
        <param name="train_image_folder" type="str" value="$(arg train_image_path)" />
        <param name="valid_annot_folder" type="str" value="$(arg valid_annot_path)" />
        <param name="valid_image_folder" type="str" value="$(arg valid_image_path)" />
        <param name="saved_weights_name" type="str" value="$(find yolov2_ros)/weights/best_full_yolo_coco.h5" />

        <rosparam param="labels">['person', 'bicycle', 'car', 'motorcycle', 'airplane', 'bus', 'train', 'truck', 'boat', 'traffic light', 'fire hydrant', 'stop sign', 'parking meter', 'bench', 'bird', 'cat', 'dog', 'horse', 'sheep', 'cow', 'elephant', 'bear', 'zebra', 'giraffe', 'backpack', 'umbrella', 'handbag', 'tie', 'suitcase', 'frisbee', 'skis', 'snowboard', 'sports ball', 'kite', 'baseball bat', 'baseball glove', 'skateboard', 'surfboard', 'tennis racket', 'bottle', 'wine glass', 'cup', 'fork', 'knife', 'spoon', 'bowl', 'banana', 'apple', 'sandwich', 'orange', 'broccoli', 'carrot', 'hot dog', 'pizza', 'donut', 'cake', 'chair', 'couch', 'potted plant', 'bed', 'dining table', 'toilet', 'tv', 'laptop', 'mouse', 'remote', 'keyboard', 'cell phone', 'microwave', 'oven', 'toaster', 'sink', 'refrigerator', 'book', 'clock', 'vase', 'scissors', 'teddy bear', 'hair drier', 'toothbrush']</rosparam>
        <param name="train_times" type="int" value="8" />
        <param name="valid_times" type="int" value="1" />
        <param name="nb_epochs" type="int" value="50" />
        <param name="learning_rate" type="double" value="1e-4" />
        <param name="batch_size" type="int" value="16" />
        <param name="warmup_epochs" type="int" value="3" />
        <param name="object_scale" type="double" value="5.0" />
        <param name="no_object_scale" type="double" value="1.0" />
        <param name="coord_scale" type="double" value="1.0" />
        <param name="class_scale" type="double" value="1.0" />
    </node>
        
</launch>